Documentation/src/case-studies/calendar.md

```diff
@@ -0,0 +1,350 @@
+
+Sequence.Successors creates an infinite sequence of days, starting with January first of the given year. We take all the values in the same year, so this sequence should yield all the days of a year.
+
+```cs
+return Sequence.Successors(JanuaryFirst(year), NextDay)
+    .TakeWhile(SameYear(year));
+```
+
+Most of these helper function are straight forward, but SameYear might be a bit special if you havent worked with curried functions before.
+
+The function `SameYear` takes one parameter and returns a function which takes another parameter, this is also called the curried form of a function, there is also the `Functional.Curry` and `Functional.Uncurry` functions which can transform between both forms without the need to write them both. `SameYear(2000)` returns a function which always returns `true` if the Date is from the year 2000. That way of using functions might come in handy a lot more often than you think.
+
+```cs
+private static DateOnly NextDay(DateOnly day)
+    => day.AddDays(1);
+
+private static DateOnly JanuaryFirst(int year)
+    => new(year, 1, 1);
+
+private static Func<DateOnly, bool> SameYear(int year)
+    => day
+        => day.Year == year;
+```
+
+This makes the the main body of the code semantic very easy to understand. All the helper functions are trivially to understand on it's own too.
+
+## Group this into months
+
+LINQ offers you the GroupBy function which we could use easily in this case.  This way we would have 12 groups, in each group we would have all the days of one month of the given year.
+
+```cs
+return Sequence.Successors(JanuaryFirst(year), NextDay)
+    .TakeWhile(SameYear(year))
+    .GroupBy(d => d.Month)
+```
+
+There are two things we should consider here though.
+
+1.) GroupBy is like the SQL GROUP BY and can rearrange elements, and therefore is not a lazy Extension function.
+2.) GroupBy would also Group all days from a different year into the same 12 montly buckets.
+
+Often that is exactly what you want, but in this case if we think of an endless stream of days this is not what we need at all. The days do not need to be rearranged, all days in the same month are next to each other and if we find a second January, we would like to have a new 13th bucket.
+
+That is why Funcky offers the extension function `AdjacentGroupBy`. It is a fully lazy, forward only `GroupBy`. Exactly what we need here.
+
+```cs
+return Sequence.Successors(JanuaryFirst(year), NextDay)
+    .TakeWhile(SameYear(year))
+    .AdjacentGroupBy(d => d.Month);
+```
+
+Now we have an `IEnumerable<IEnumerable<DateOnly>>`, where the inner one is representing a single month.
+
+## Layout a single month
+
+Since we have now all the days of the same month, we want to create a layout for a single month out of these days.
+
+Currently we have a sequence of months. To transform the inner element, from a sequence of days into something else, we need to make a projection which is done with Select in C#.
```

Comment:
add single ticks around Select

Reply:
Fixed in 9e90ff193883aab4acb6fbb4758a68b84cac8a23